hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/AbstractBatchEntitySelectFetchInitializer.java

```diff
@@ -136,10 +158,13 @@ public void registerResolutionListener(Consumer<Object> listener) {
 		}
 	}
 
-	protected static Object loadInstance(
+	protected Object loadInstance(
 			EntityKey entityKey,
 			ToOneAttributeMapping referencedModelPart,
 			SharedSessionContractImplementor session) {
+		if ( entityInstance != null ) {
+			return entityInstance;
+		}
```

Comment:
This change doesn't seem correct or sensible to me, because this method is called at the end of `ResultSet` processing so the `entityInstance` would AFAICT always be `null`.